### PR TITLE
refactor: xId to apiKey

### PR DESCRIPTION
**Pull request by tmm**
## Description

Switch `alchemyProvider` and `infuraProvider` to use a generic `apiKey` configuration option instead of `alchemyId` and `infuraId`.

```diff
import { alchemyProvider } from '@wagmi/core/providers/alchemy'
import { infuraProvider } from '@wagmi/core/providers/infura'

alchemyProvider({
-  alchemyId: 'yourAlchemyApiKey',
+  apiKey: 'yourAlchemyApiKey',
})

infuraProvider({
-  infuraId: 'yourInfuraApiKey',
+  apiKey: 'yourInfuraApiKey',
})
```

## Additional Information

- [x] I read the [contributing docs](/wagmi-dev/wagmi/blob/main/.github/CONTRIBUTING.md) (if this is your first contribution)

Your ENS/address: awkweb.eth

### CODE REVIEW

**Comment by subject026** [comment]
💯

File: .changeset/brave-penguins-sniff.md [hunk 1]
```diff
@@ -4,10 +4,10 @@
 
 **Breaking:** The `useSendTransaction` hook's `data` now returns an object only consisting of `hash` & `wait`, and not the full [`TransactionResponse`](https://docs.ethers.io/v5/api/providers/types/#providers-TransactionResponse).
 
-If you require the full `TransactionResponse`, you can use `useWaitForTransaction` with `confirmations: 0`:
+If you require the full `TransactionResponse`, you can use `useTransaction`:
```
Comment (jxom) [inline .changeset/brave-penguins-sniff.md:7]: Is this meant to be in this PR?

**Review by jxom — APPROVED** [review]
Got some `useTransaction` leakage, but looks good!

File: .changeset/brave-penguins-sniff.md (hunk 1 above)
Comment (tmm) [reply .changeset/brave-penguins-sniff.md:7]: will add that over to #759

**Pull request merged**